debezium-core/src/main/java/io/debezium/pipeline/AbstractChangeRecordEmitter.java

```diff
@@ -24,10 +25,13 @@
     private final OffsetContext offsetContext;
     private final Clock clock;
 
-    public AbstractChangeRecordEmitter(P partition, OffsetContext offsetContext, Clock clock) {
+    private final CommonConnectorConfig connectorConfig;
+
+    public AbstractChangeRecordEmitter(P partition, OffsetContext offsetContext, Clock clock, CommonConnectorConfig connectorConfig) {
```

Comment:
I'm not a fan of pushing `CommonConnectorConfig` and the `skipMessagesWithoutChange` method up this far into the class hierarchy as it forces MongoDB to explicitly return false, making most of these changes at this level in the class hierarchy a bit superfluous.

I'd prefer you only pass `CommonConnectorConfig` up to the `RelationalChangeRecordEmitter` and simply expose the `skipMessagesWithoutChange` there, which gives a clear indication that this is a relational connector-only feature as it isn't exposed at the top of the class hierarchy that both relational and non-relational connectors are based from.

Reply:
After reviewing the Jira issue, it seems we may plan to extend this to MongoDB, and in that case, this makes sense.

Reply:
Marking this as resolved